pkg/controller/controller_instance.go

```diff
@@ -250,19 +297,30 @@ func (c *controller) reconcileInstance(instance *v1alpha1.Instance) error {
 
 	glog.V(4).Infof("Adding/Updating Instance %v/%v", instance.Namespace, instance.Name)
 
+	// we will definitely update the instance's status - make a deep copy now
+	// for use later in this method.
+	clone, err := api.Scheme.DeepCopy(instance)
```

Comment:
What are your thoughts on moving this into the special case async section? If I haven't missed anything, that's the only place where multiple fields are being edited and require a DeepCopy to get a multiple field atomic update (perhaps that's not the reason for doing the clone?) Would require a little refactoring to make it work.

Should controller.go also use the same set/update pattern for the instance condition too?